### PR TITLE
feature/AGCMD-5297 - replaced then-redis and upgraded to node v8.4.0

### PR DESCRIPTION
**Pull request by andreymoser**

### CODE REVIEW

File: test/redisOpenAM.spec.js [hunk 1]
```diff
@@ -24,14 +27,14 @@ var mockToken = 'db6e6138-3f53-4065-9610-25022a175516';
...
+    var redis;
 
     before(function configRedisDb() {
-        redisDb = Redis.createClient();
+        redis = Redis.createClient();
     });
 
     beforeEach(function setupMockOptions() {
-        redisDb.flushall();
+        redis.flushallAsync();
```
Comment (ssebro) [inline test/redisOpenAM.spec.js]: This may cause you bugs. You probably want to still synchronously flushall

Comment (ssebro) [reply test/redisOpenAM.spec.js]: Or i *think* you could just return the promise. that should help.

**Review by ssebro — COMMENTED** [review]
Everything looks good but that comment. Fix it and we're good to go - good work!

**Review by hvhallmann — COMMENTED** [review]
Sounds good to me. Go for it!

**Pull request merged**